src/types/mod.rs

```diff
@@ -774,3 +808,171 @@ impl<T: Object> ValueInternal for JsFunction<T> {
         unsafe { neon_runtime::tag::is_function(env.to_raw(), other.to_raw()) }
     }
 }
+
+/// A builder for making a JavaScript function call (e.g., `parseInt("42")`).
+///
+/// The builder methods make it convenient to assemble the call from parts:
+/// ```
+/// # use neon::prelude::*;
+/// # fn foo(mut cx: FunctionContext) -> JsResult<JsNumber> {
+/// # let global = cx.global();
+/// # let parseInt = global.get(&mut cx, "parseInt")?;
+/// # let parseInt: Handle<JsFunction> = parseInt.downcast_or_throw(&mut cx)?;
+/// let x: Handle<JsNumber> = parseInt
+///     .arg(cx.string("42"))
+///     .call(&mut cx)?;
+/// # Ok(x)
+/// # }
+/// ```
+#[derive(Clone)]
+pub struct FunctionCall<'a> {
+    callee: Handle<'a, JsFunction>,
+    this: Handle<'a, JsValue>,
+    args: Vec<Handle<'a, JsValue>>,
```

Comment:
We could consider [`tinyvec`](https://docs.rs/tinyvec/1.4.0/tinyvec/index.html) to try to avoid heap allocations.

Reply:
I've stuck with SmallVec for the moment but I can migrate to TinyVec a little later. Getting CI to pass again ATM…